### PR TITLE
Fix ipv6 related issues

### PR DESCRIPTION
**Pull request by EdDev**
Fix several issues related to IPv6 in both the production code and integration tests.

### CODE REVIEW

File: libnmstate/netapplier.py [hunk 1]
```diff
@@ -150,6 +152,13 @@ def _apply_ifaces_state(
         raise NmstatePermissionError('Error creating a check point')
     except nm.checkpoint.NMCheckPointCreationError:
         raise NmstateConflictError('Error creating a check point')
+    except NmstateError:
+        # Assume rollback occured, revert IPv6 stack state.
+        # Checkpoint rollback is async, there is a need to wait for it to
+        # finish before proceeding with other actions.
+        time.sleep(5)
```
Comment (cathay4t) [inline libnmstate/netapplier.py]: Maybe we need a `TODO` tag here to indicate we should replace sleep with rollback wait function once we have.

File: libnmstate/netapplier.py [hunk 2]
```diff
@@ -150,6 +152,13 @@ def _apply_ifaces_state(
         raise NmstatePermissionError('Error creating a check point')
     except nm.checkpoint.NMCheckPointCreationError:
         raise NmstateConflictError('Error creating a check point')
+    except NmstateError:
+        # Assume rollback occured, revert IPv6 stack state.
```
Comment (maiqueb) [inline libnmstate/netapplier.py]: I'm confused with the wording here; does this roll back or not ?

If not I agree whit @cathay4t comments, if so, I don't understand them.

File: tests/integration/dhcp_test.py [hunk 3]
```diff
@@ -683,6 +678,17 @@ def test_slave_ipaddr_learned_via_dhcp_added_as_static_to_linux_bridge(
     assertlib.assert_state(bridge_desired_state)
 
 
+@pytest.mark.xfail(raises=NmstateNotImplementedError)
```
Comment (maiqueb) [inline tests/integration/dhcp_test.py]: I think re-ordering the test is unrelated to the other issue fixed here, and that it deserves a separate commit.

Not that picky though.

File: libnmstate/netapplier.py (hunk 2 above)
Comment (cathay4t) [reply libnmstate/netapplier.py]: @maiqueb Currently, we just send a request to NM asking a rollback when failure. But we don't check on whether rollback finished. Hence @EdDev put a `sleep(5)` for that.

Once we have rollback wait support, we can remove this sleep.

File: libnmstate/netapplier.py (hunk 1 above)
Comment (EdDev) [reply libnmstate/netapplier.py]: I replaced them after your findings of the test fail cause.
Basically, I used your #449 idea with some changes.

Comment (EdDev) [reply libnmstate/netapplier.py]: Done

File: tests/integration/dhcp_test.py (hunk 3 above)
Comment (EdDev) [reply tests/integration/dhcp_test.py]: It is related because without it all tests below it fail. Relocating it in its own commit is possible, but then it will have no context/reason.

**Comment by EdDev** [comment]
The tests are passing on CentOS7 but not on Fedora. The failure is always on the nameservers assertion. I cannot figure out what is wrong.

**Comment by maiqueb** [comment]
> The tests are passing on CentOS7 but not on Fedora. The failure is always on the nameservers assertion. I cannot figure out what is wrong.

The error you're facing is that you have extra nameserver(s) in the *running* section , right ? 

I've seen that issue also in the vdsm static_ip tests, for the ```TestNetworkIPDefaultGateway```. I guess it comes from the nameserver libvirt configures on the interface that gets an IP address via dhcp, on my VM. Meaning, meaning that depending on your config, you could be seeing different symptoms.

I got around it by removing that nameserver from the configuration (actually I went neanderthal on it and just ```echo "" > /etc/resolv.conf ``` . 

I think we could / should **not** report back in the 'running' field nameservers / routes that were not configured. *Or*, at least have an option to do that for all entities having the *config*/*running* logical separation. It would be kind of a simpler implementation of the **owner** field - i.e. anything that is **not** listed in the 'config' section is **not** owned by nmstate.

**Comment by cathay4t** [comment]
@maiqueb 
> I think we could / should **not** report back in the 'running' field nameservers / routes that were not configured. _Or_, at least have an option to do that for all entities having the _config_/_running_ logical separation. It would be kind of a simpler implementation of the **owner** field - i.e. anything that is **not** listed in the 'config' section is **not** owned by nmstate.

For DHCP, we cannot determine which nameserver is owned.

**Comment by maiqueb** [comment]
> @maiqueb
> 
> > I think we could / should **not** report back in the 'running' field nameservers / routes that were not configured. _Or_, at least have an option to do that for all entities having the _config_/_running_ logical separation. It would be kind of a simpler implementation of the **owner** field - i.e. anything that is **not** listed in the 'config' section is **not** owned by nmstate.
> 
> For DHCP, we cannot determine which nameserver is owned.

Hm, can't we list the nameservers that are **only** listed in the *running* section ? Those are the one's not nmstate related.

**Comment by cathay4t** [comment]
> > @maiqueb
> > > I think we could / should **not** report back in the 'running' field nameservers / routes that were not configured. _Or_, at least have an option to do that for all entities having the _config_/_running_ logical separation. It would be kind of a simpler implementation of the **owner** field - i.e. anything that is **not** listed in the 'config' section is **not** owned by nmstate.
> > 
> > 
> > For DHCP, we cannot determine which nameserver is owned.
> 
> Hm, can't we list the nameservers that are **only** listed in the _running_ section ? Those are the one's not nmstate related.

For this PR, root cause is the [reapply bug of NM][1], if you want to discuss the DNS schema, we
can move it to github issue or github PR or jira card.

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1734470

**Comment by EdDev** [comment]
Depends on #453, #454 .

File: libnmstate/netapplier.py (hunk 2 above)
Comment (maiqueb) [reply libnmstate/netapplier.py]: Ah ok. Got it, tks.

File: tests/integration/dhcp_test.py (hunk 3 above)
Comment (maiqueb) [reply tests/integration/dhcp_test.py]: Fair enough, as I said, won't block this because of that.

**Comment by maiqueb** [comment]
Shouldn't CI pass now that you've got the other 2 PRs (#453, #454 ) rebased here ?

**Comment by EdDev** [comment]
> Shouldn't CI pass now that you've got the other 2 PRs (#453, #454 ) rebased here ?

We fail randomly on OVS tests. Gris already has a change that takes them out using `xfail`.

**Comment by maiqueb** [comment]
> > Shouldn't CI pass now that you've got the other 2 PRs (#453, #454 ) rebased here ?
> 
> We fail randomly on OVS tests. @cathay4t already has a change that takes them out using `xfail`.

Right, my bad; didn't look at the failed test list, sorry for that.
